eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/source/protocol/impl/GitHubProtocol.java

```diff
@@ -0,0 +1,215 @@
+
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.handler.BodyHandler;
+
+import com.alibaba.fastjson2.JSON;
+import com.alibaba.fastjson2.JSONWriter.Feature;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * GitHub Protocol. This class represents the GitHub webhook protocol.
+ */
+@Slf4j
+public class GitHubProtocol implements Protocol {
+
+    // Protocol name
+    public static final String PROTOCOL_NAME = "GitHub";
+
+    private static final String H_MAC_SHA_265 = "HmacSHA256";
+
+    private static final String SECRET_KEY = "secret";
+
+    private String secret;
+
+
+    /**
+     * Initialize the protocol.
+     *
+     * @param sourceConnectorConfig source connector config
+     */
+    @Override
+    public void initialize(SourceConnectorConfig sourceConnectorConfig) {
+        // Initialize the protocol
+        Map<String, String> extraConfig = sourceConnectorConfig.getExtraConfig();
+        this.secret = extraConfig.get(SECRET_KEY);
+        if (StringUtils.isBlank(this.secret)) {
+            throw new EventMeshException("The secret is required for GitHub protocol.");
+        }
+    }
+
+    /**
+     * Handle the protocol message for GitHub.
+     *
+     * @param route     route
+     * @param boundedQueue queue info
+     */
+    @Override
+    public void setHandler(Route route, BoundedConcurrentQueue<Object> boundedQueue) {
+        route.method(HttpMethod.POST)
+            .handler(BodyHandler.create())
+            .handler(ctx -> {
+                // Get the payload
+                String payload = ctx.body().asString(Constants.DEFAULT_CHARSET.toString());
+
+                // Get the headers
+                MultiMap headers = ctx.request().headers();
+                // validate the signature
+                String signature = headers.get(WebhookConstants.GITHUB_SIGNATURE_256);
+                if (BooleanUtils.isFalse(validateSignature(signature, payload, secret))) {
+                    String errorMsg = String.format("signature is invalid, please check the secret. received signature: %s", signature);
+                    log.error(errorMsg);
+
+                    WebhookResponse response = WebhookResponse.builder()
+                        .msg(errorMsg)
+                        .handleTime(LocalDateTime.now())
+                        .build();
+
+                    // Return 400 Bad Request
+                    ctx.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
+                        .send(JSON.toJSONString(response, Feature.WriteMapNullValue));
+                    return;
```

Comment:
You can throw an exception here and let failureHandler to response.

Reply:
Good idea, I will modify it

Reply:
Solved at  https://github.com/apache/eventmesh/pull/4913/commits/b175fd22b91aab83b3adb22131306f0b0df29de3